programs/mpl-core/src/plugins/lifecycle.rs

```diff
@@ -904,11 +931,19 @@ pub(crate) fn validate_external_plugin_checks<'a>(
 
             let result = external_plugin_validate_fp(
                 &ExternalPlugin::load(account, external_registry_record.offset)?,
-                &ctx,
+                &validation_ctx,
             )?;
             match result {
-                ValidationResult::Rejected => return Ok(ValidationResult::Rejected),
-                ValidationResult::Approved => approved = true,
+                ValidationResult::Rejected => {
+                    if check_result.can_reject() {
+                        return Ok(ValidationResult::Rejected);
+                    }
+                }
+                ValidationResult::Approved => {
+                    if check_result.can_approve() {
+                        approved = true;
+                    }
+                }
```

Comment:
### Design discussion

Note I'm making sure the external plugin actually has been configured to have authority to approve or reject before accepting the result from a low-level call to a  `PluginValidation` trait method such as `validate_transfer`.   This also occurs in `create` and `create_collection`.

From a SW design standpoint this level seems correct because it is where we already are managing the external check results.

However, from a mistake-proof standpoint, I kind of want to put the `external_check_result` into the validation struct so that at the lowest level (i.e. the call to `PluginValidation::validate_transfer()`, etc.) we do NOT return any `ValidationResult` the plugin is not supposed to return.

I think this mistake-proof risk somewhat extends to first party plugins as well, but there its more just a code consistency issue.  For example if we configure a plugin so `check_transfer` returns `CheckResult::CanApprove`, but in practice `validate_transfer` could return `ValidationResult::Rejected`.

But overall this handling is more important for external plugins since they are meant to be configured up front.

Reply:
OK, looking at this more I see your point. I think you're right that it makes sense to check at the lowest level so we don't accidentally mix check results. Maybe we can just add the external records to the context?